akka-remote/src/test/scala/akka/remote/artery/tcp/ssl/RotatingKeysSSLEngineSpec.scala

```diff
@@ -0,0 +1,152 @@
+        rotating-keys-engine {
+          key-file = ${getClass.getClassLoader.getResource("ssl/node.example.com.pem").getPath}
+          cert-file = ${getClass.getClassLoader.getResource("ssl/node.example.com.crt").getPath}
+          ca-cert-file = ${getClass.getClassLoader.getResource("ssl/exampleca.crt").getPath}
+          ssl-context-cache-ttl = ${cacheTtlInSeconds}s
+        }
+      }
+    """
+  }
+  val config: Config = ConfigFactory.parseString(configStr)
+
+}
+
+// In this test each system reads keys/certs from a different temporal folder to control
+// which system gets keys rotated.
+class RotatingKeysSSLEngineSpec
+    extends ArteryMultiNodeSpec(RotatingKeysSSLEngineSpec.config.withFallback(TlsTcpSpec.config))
+    with ImplicitSender {
+  "Artery with TLS/TCP with RotatingKeysSSLEngine" must {
+    "rotate keys and rebuild the SSLContext" in {
+      if (!arteryTcpTlsEnabled())
+        pending
+
+      // an initial connection between sysA (from the testkit) and sysB
+      // to get sysB up and runnning
+      val remoteSysB = new RemoteSystem("systemB", newRemoteSystem, address)
+      remoteSysB.actorSystem.actorOf(TestActors.echoActorProps, "echoB")
+      val pathEchoB = remoteSysB.rootActorPath / "user" / "echoB"
+      system.actorSelection(pathEchoB) ! Identify(pathEchoB.name)
+      val echoBRef: ActorRef = expectMsgType[ActorIdentity].ref.get
+      echoBRef ! "ping-1"
+      expectMsg("ping-1")
+
+      remoteSysB.sslProviderServerProbe.expectMsg("createServerSSLEngine")
+      remoteSysB.sslProviderClientProbe.expectMsg("createClientSSLEngine")
+      val before = remoteSysB.sslContextRef.get()
+
+      // sleep to force the cache in sysB's instance to expire
+      Thread.sleep((RotatingKeysSSLEngineSpec.cacheTtlInSeconds + 1) * 1000)
+      // Connect system C to system B because I can't get a reference to the SSLContext in system A
+      val remoteSysC = new RemoteSystem("systemC", newRemoteSystem, address)
+      remoteSysC.actorSystem.actorOf(TestActors.echoActorProps, "echoC")
+      val pathEchoC = remoteSysC.rootActorPath / "user" / "echoC"
+      val senderOnB = TestProbe()(remoteSysB.actorSystem)
+      remoteSysB.actorSystem.actorSelection(pathEchoC).tell(Identify(pathEchoC.name), senderOnB.ref)
+      val echoCRef: ActorRef = senderOnB.expectMsgType[ActorIdentity].ref.get
+      echoCRef.tell("ping-1", senderOnB.ref)
+      senderOnB.expectMsg("ping-1")
+
+      // Expect both sysB and sysC had seldom requests for connections.
+      remoteSysC.sslProviderServerProbe.expectMsg("createServerSSLEngine")
+      remoteSysC.sslProviderClientProbe.expectMsg("createClientSSLEngine")
+      remoteSysB.sslProviderServerProbe.expectMsg("createServerSSLEngine")
+      remoteSysB.sslProviderClientProbe.expectMsg("createClientSSLEngine")
+
+      // the SSLContext references on sysB should differ
+      val after = remoteSysB.sslContextRef.get()
+      before shouldNot be(after)
+
+    }
```

Comment:
Note: this tests only asserts a new `SSLContext` is built. It doesn't test keys on the FS are re-read yet.